infra/modules/providers/azure/flux/main.tf

```diff
@@ -15,12 +15,68 @@
 resource "null_resource" "deploy_flux" {
   count = var.enable_flux ? 1 : 0
 
-  provisioner "local-exec" {
-    command = "echo 'Need to use this var so terraform waits for kubeconfig ' ${var.kubeconfig_complete};KUBECONFIG=${var.output_directory}/${var.kubeconfig_filename} ${path.module}/deploy_flux.sh -b '${var.gitops_url_branch}' -f '${var.flux_repo_url}' -g '${var.gitops_ssh_url}' -k '${var.gitops_ssh_key_path}' -d '${var.flux_clone_dir}' -c '${var.gitops_poll_interval}' -l '${var.gitops_label}' -e '${var.gitops_path}' -s '${var.acr_enabled}' -r '${var.flux_image_repository}' -t '${var.flux_image_tag}' -z '${var.gc_enabled}'"
-  }
-
   triggers = {
     enable_flux   = var.enable_flux
     flux_recreate = var.flux_recreate
   }
-}
\ No newline at end of file
+
+  provisioner "local-exec" {
+    command = <<EOT
+     chmod 744 ${path.module}/deploy_flux.sh
+     ls -l ${path.module}
+     echo 'Need to use this var so terraform waits for kubeconfig ' ${var.kubeconfig_complete}
+     KUBECONFIG=${var.output_directory}/${var.kubeconfig_filename} ${path.module}/deploy_flux.sh \
+      -b '${var.gitops_url_branch}' \
+      -f '${var.flux_repo_url}' \
+      -g '${var.gitops_ssh_url}' \
+      -k '${var.gitops_ssh_key_path}' \
+      -d '${var.flux_clone_dir}' \
+      -c '${var.gitops_poll_interval}' \
+      -l '${var.gitops_label}' \
+      -e '${var.gitops_path}' \
+      -s '${var.acr_enabled}' \
+      -r '${var.flux_image_repository}' \
+      -t '${var.flux_image_tag}' \
+      -z '${var.gc_enabled}'
+EOT
+  }
+}
+
+
+# resource "null_resource" "deploy_flux" {
```

Comment:
What are these commented code for - future work?

Reply:
There was a lot of troubleshooting to find a pipeline only error on this PR for properly deploying flux that scripts worked fine locally but would fail when running in the build agent due to some differences in the host os and bash versions.